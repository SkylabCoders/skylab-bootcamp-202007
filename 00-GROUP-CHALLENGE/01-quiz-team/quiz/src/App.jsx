import React from 'react';
import './css/app.scss';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import PageNotFound from './components/PageNotFound';
import Header from './components/Header';
import Footer from './components/Footer';
import LeaderBoard from './components/LeaderBoard';
import LoginScreen from './components/LoginScreen';
import Dashboard from './components/Dashboard';
import Question from './components/Question';
import Game from './components/Game';
import Answer from './components/Answer';
import * as ROUTES from './config/routes';

function App() {
	return (
		<div className="app">
			<Router>
			<Header />
			<Switch>
				<Route path={ROUTES.HOME} exact component={Dashboard} />
				<Route path={ROUTES.LEADERBOARD} component={LeaderBoard} />
				<Route path={ROUTES.LOGIN} component={LoginScreen} />
				<Route path={ROUTES.QUESTION} component={Question} />
				<Route path={ROUTES.THEMES_WILDCARD} component={Game} />
				<Route path={ROUTES.ANSWER} component={Answer} />
				<Route component={PageNotFound} />
			</Switch>
			<Footer />
			</Router>
		</div>
	);
}

export default App;
