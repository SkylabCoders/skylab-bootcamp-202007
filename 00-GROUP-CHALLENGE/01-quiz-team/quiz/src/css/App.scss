.app {
  width: 100%;
}

/*# sourceMappingURL=App.scss.map */
