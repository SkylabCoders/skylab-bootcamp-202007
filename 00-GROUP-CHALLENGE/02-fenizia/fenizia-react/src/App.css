* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Lato', sans-serif;
}


.margin-body {
  padding-top: 60px;
  min-height: calc(100vh - 140px - 0px); 
}
