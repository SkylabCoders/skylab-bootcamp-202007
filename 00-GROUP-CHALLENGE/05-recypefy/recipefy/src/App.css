* {
  margin: 0;
  padding: 0;
  font-family: Arial, Helvetica, sans-serif !important;
}

#root {
  width: 100%;
}

body {
  background-color: #ffff;
  width: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
}

a {
  text-decoration: none;
  color: black;
}

h2 {
  font-family: Georgia, 'Times New Roman', Times, serif;
  color: black;
  font-size: 24px;
  text-transform: uppercase;
}

h3 {
  font-family: Georgia, 'Times New Roman', Times, serif;
  color: #01010e;
}

h4 {
  font-family: Georgia, 'Times New Roman', Times, serif;
  color: #010113;
}

p {
  font-family: Georgia, 'Times New Roman', Times, serif;
  color: #010111;
}

.body__list {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  text-align: center;
  -webkit-box-align: stretch;
      -ms-flex-align: stretch;
          align-items: stretch;
}

.body__list h2 {
  padding: 15px;
}

.body__list ul {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
      -ms-flex-direction: row;
          flex-direction: row;
  -ms-flex-wrap: wrap;
      flex-wrap: wrap;
  -webkit-box-pack: space-evenly;
      -ms-flex-pack: space-evenly;
          justify-content: space-evenly;
  width: auto;
}

.body__box {
  height: 35vh;
  height: auto;
  width: 90%;
  background-color: #f8f8f0;
  margin: 5% 2% 5% 2%;
  padding: 3%;
  border-radius: 25px;
  -webkit-box-shadow: 10px 13px 8px -5px rgba(0, 0, 0, 0.08);
  box-shadow: 10px 13px 8px -5px rgba(0, 0, 0, 0.08);
}

.body__box .recipe__text {
  width: 50%;
  min-width: 250px;
  height: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: space-evenly;
      -ms-flex-pack: space-evenly;
          justify-content: space-evenly;
}

.body__box .recipe__text--title {
  margin-top: 20px;
  margin-bottom: 20px;
  max-width: 40 vw;
  color: #180101;
  margin-left: 20px;
  font-size: 17px;
}

.body__box .recipe__image {
  border-radius: 8%;
  width: 100%;
  height: auto;
  margin-right: 5px;
}

.body__box--recipe-details {
  background-color: #f8f8f0;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
}

.body__box--recipe-details a {
  width: 100%;
  margin: 15px;
}

.preferences__icon {
  width: 30px;
  height: 30px;
  border-radius: 100%;
  margin-left: 5px;
  margin-right: 5px;
  margin-top: 5px;
}

.preference__peanut-free {
  -webkit-filter: invert(51%) sepia(96%) saturate(461%) hue-rotate(329deg) brightness(99%) contrast(94%);
          filter: invert(51%) sepia(96%) saturate(461%) hue-rotate(329deg) brightness(99%) contrast(94%);
}

.preference__tree-nut-free {
  -webkit-filter: invert(70%) sepia(11%) saturate(3163%) hue-rotate(69deg) brightness(95%) contrast(66%);
          filter: invert(70%) sepia(11%) saturate(3163%) hue-rotate(69deg) brightness(95%) contrast(66%);
}

.preference__alcohol-free {
  -webkit-filter: invert(10%) sepia(57%) saturate(3660%) hue-rotate(226deg) brightness(81%) contrast(92%);
          filter: invert(10%) sepia(57%) saturate(3660%) hue-rotate(226deg) brightness(81%) contrast(92%);
}

.box__title {
  margin-top: 20px;
  margin-left: 20px;
  color: #010113;
}

.recipe__text--preferences {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -ms-flex-wrap: wrap;
      flex-wrap: wrap;
  -ms-flex-pack: distribute;
      justify-content: space-around;
}

.loading {
  width: 100px;
  height: 100px;
}

@media screen and (min-width: 600px) {
  .body__box .recipe__image {
    margin-right: 5px;
    max-height: 30vw;
    width: auto;
  }
  .body__box--recipe-details {
    -webkit-box-orient: horizontal;
    -webkit-box-direction: normal;
        -ms-flex-direction: row;
            flex-direction: row;
  }
  .body__box--recipe-details a {
    text-align-last: center;
  }
}

@media screen and (min-width: 1024px) {
  .Main-page .body__box {
    width: 62%;
    margin: 2%;
    margin-top: 5%;
  }
  .body-content {
    position: relative;
    width: 80%;
    margin: auto;
  }
}
/*# sourceMappingURL=App.css.map */