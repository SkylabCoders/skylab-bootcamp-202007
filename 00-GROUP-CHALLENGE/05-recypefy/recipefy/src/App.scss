* {
	margin: 0;
	padding: 0;
	font-family: Arial, Helvetica, sans-serif !important;
}

#root {
	width: 100%;
}

body {
	background-color: #ffff;
	width: 100%;
	display: flex;
}

a {
	text-decoration: none;
	color: black;
}

h2 {
	font-family: Georgia, 'Times New Roman', Times, serif;
	color: rgb(0, 0, 0);
	font-size: 24px;
	text-transform: uppercase;
}

h3 {
	font-family: Georgia, 'Times New Roman', Times, serif;
	color: rgb(1, 1, 14);
}

h4 {
	font-family: Georgia, 'Times New Roman', Times, serif;
	color: rgb(1, 1, 19);
}

p {
	font-family: Georgia, 'Times New Roman', Times, serif;
	color: rgb(1, 1, 17);
}

.body__list {
	display: flex;
	flex-direction: column;
	text-align: center;

	align-items: stretch;

	h2 {
		padding: 15px;
	}

	ul {
		display: flex;
		flex-direction: row;
		flex-wrap: wrap;
		justify-content: space-evenly;
		width: auto;
	}
}

.body__box {
	height: 35vh;
	height: auto;
	width: 90%;
	background-color: #f8f8f0;
	margin: 5% 2% 5% 2%;
	padding: 3%;
	border-radius: 25px;
	-webkit-box-shadow: 10px 13px 8px -5px rgba(0, 0, 0, 0.08);
	-moz-box-shadow: 10px 13px 8px -5px rgba(0, 0, 0, 0.08);
	box-shadow: 10px 13px 8px -5px rgba(0, 0, 0, 0.08);

	.recipe__text {
		width: 50%;
		min-width: 250px;
		height: 100%;
		display: flex;
		flex-direction: column;
		align-items: center;
		justify-content: space-evenly;
	}

	.recipe__text--title {
		margin-top: 20px;
		margin-bottom: 20px;
		max-width: 40 vw;
		color: rgb(24, 1, 1);
		margin-left: 20px;
		font-size: 17px;
	}

	.recipe__image {
		border-radius: 8%;
		width: 100%;
		height: auto;
		margin-right: 5px;
	}
}

.body__box--recipe-details {
	background-color: #f8f8f0;
	display: flex;
	align-items: center;
	flex-direction: column;

	a {
		width: 100%;
		margin: 15px;
	}
}

.preferences__icon {
	width: 30px;
	height: 30px;
	border-radius: 100%;
	margin-left: 5px;
	margin-right: 5px;
	margin-top: 5px;
}

.preference__peanut-free {
	filter: invert(51%) sepia(96%) saturate(461%) hue-rotate(329deg) brightness(99%) contrast(94%);
}

.preference__tree-nut-free {
	filter: invert(70%) sepia(11%) saturate(3163%) hue-rotate(69deg) brightness(95%) contrast(66%);
}

.preference__alcohol-free {
	filter: invert(10%) sepia(57%) saturate(3660%) hue-rotate(226deg) brightness(81%) contrast(92%);
}

.box__title {
	margin-top: 20px;
	margin-left: 20px;
	color: rgb(1, 1, 19);
}

.recipe__text--preferences {
	display: flex;
	flex-wrap: wrap;
	justify-content: space-around;
}

.loading {
	width: 100px;
	height: 100px;
}

@media screen and (min-width: 600px) {
	.body__box {
		.recipe__image {
			margin-right: 5px;
			max-height: 30vw;
			width: auto;
		}
	}

	.body__box--recipe-details {
		flex-direction: row;

		a {
			text-align-last: center;
		}
	}
}

@media screen and (min-width: 1024px) {
	.Main-page {
		.body__box {
			width: 62%;
			margin: 2%;
			margin-top: 5%;
		}
	}

	.body-content {
		position: relative;
		width: 80%;
		margin: auto;
	}
}