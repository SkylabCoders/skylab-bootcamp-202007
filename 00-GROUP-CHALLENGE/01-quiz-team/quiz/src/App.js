import React from 'react';
import './css/App.css';
import { Route, Switch } from 'react-router-dom';
import PageNotFound from './components/PageNotFound'
import Home from './components/Home'
import Header from './components/Header'
import Footer from './components/Footer'
import LeaderBoard from './components/LeaderBoard'
import LoginScreen from './components/LoginScreen'
import Dashboard from './components/Dashboard'


function App() {
	return (
		<div className="app">
			<Header />
			<Switch>
				<Route path="/" exact component={Dashboard} />
				<Route path="/leaderboard" exact component={LeaderBoard} />
				<Route path="/login" exact component={LoginScreen} />
				<Route component={PageNotFound} />
			</Switch>
			<Footer />
		</div>
	);
}

export default App;
