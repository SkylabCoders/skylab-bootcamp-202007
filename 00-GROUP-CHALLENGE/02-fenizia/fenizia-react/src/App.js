import React from "react";
import { Route, Switch } from "react-router-dom";
import "./App.css";
import Header from "./components/Header";
import Dashboard from "./components/Dashboard";
import Footer from "./components/Footer";
import Profile from "./components/Profile";
import Login from "./components/Login";
import ProductList from "./components/ProductList";
import Cart from "./components/Cart";
import Register from "./components/Register";
import About from "./components/About";


function App(props) {
  return (
    <div className="container">
      <Header />
      
      <div className="margin-body">
        <Switch>
          <Route path="/" exact component={Dashboard} />
          <Route path="/profile" exact component={Profile} />
          <Route path="/register" exact component={Register} />
          <Route path="/cart" exact component={Cart} />
          <Route path="/login" exact component={Login} />
          <Route path="/cart" exact component={Cart} />
          <Route path="/register" exact component={Register} />
          <Route path="/sobre-nosotros" exact component={About} />
          <Route path="/:genre" exact component={ProductList} />
          <Route path="/infantil" exact component={ProductList} />
          <Route path="/romantico" exact component={ProductList} />
          <Route path="/historia" exact component={ProductList} />
          <Route path="/cocina" exact component={ProductList} />
          <Route path="/ensayo" exact component={ProductList} />
          <Route path="/thriller" exact component={ProductList} />
          <Route path="/todos" exact component={ProductList} />
          <Route path="/novedades" exact component={ProductList} />
          <Route path="/superventas" exact component={ProductList} />

          {/* <Route component={PageNotFound} /> */}
        </Switch>
      </div>
    
      <Footer />
    </div>
  );
}

export default App;
