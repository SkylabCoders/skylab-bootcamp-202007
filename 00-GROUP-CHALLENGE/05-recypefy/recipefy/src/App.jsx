import React, { useState, useEffect } from 'react';
import { Route, Switch } from 'react-router-dom';
import Footer from './components/Footer';
import Header from './components/Header';
import Main from './components/Main';
import RecipePage from './components/RecipePage';
import SearchResult from './components/SearchResult';
import './App.scss';
import Login from './components/Login';
import authStore from './stores/AuthStore';

function App() {
	const [userLogued, setUserLogued] = useState(false);

	useEffect(() => {
		authStore.addChangeListener(onChange);
		return () => authStore.removeChangeListener(onChange);
	}, [userLogued]);

	function onChange() {
		setUserLogued(authStore.isLogged());
	}
	return (
		<>
			{' '}
			{userLogued && <Header />}
			<Switch>
				<Route path="/main" exact component={Main} />
				<Route path="/recipe-page/:recipeId" component={RecipePage} />
				<Route path="/search-result" component={SearchResult} />
				<Route path="/" component={Login} />
			</Switch>
			<Footer />
		</>
	);
}

export default App;
