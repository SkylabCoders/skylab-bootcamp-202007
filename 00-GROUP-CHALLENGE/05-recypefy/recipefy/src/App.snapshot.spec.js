import renderer from 'react-test-renderer';
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import App from './App';

describe('App snapshot', () => {
	const tree = renderer.create(
		<Router>
			<App />
		</Router>
	);

	it('should match', () => {
		expect(tree.toJSON()).toMatchSnapshot();
	});
});
