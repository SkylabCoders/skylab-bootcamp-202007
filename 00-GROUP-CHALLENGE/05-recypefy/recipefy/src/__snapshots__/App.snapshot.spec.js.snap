// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot should match 1`] = `
Array [
  " ",
  <div
    className="login__container"
  >
    <div
      className="login__box"
    >
      <img
        className="login__logo--size"
        src="https://image.flaticon.com/icons/svg/770/770906.svg"
      />
      <h2
        className="login__title"
      >
        Login with
      </h2>
      <form />
      <a
        onClick={[Function]}
      >
        EMAIL
      </a>
      <a
        onClick={[Function]}
      >
        GOOGLE
      </a>
      <a
        onClick={[Function]}
      >
        ANON
      </a>
      <h2
        className="login__title"
      >
        OR
      </h2>
      <a
        onClick={[Function]}
      >
        REGISTER
      </a>
    </div>
  </div>,
  <footer>
    <div
      className="developers"
    >
      <div
        className="cards__container"
      >
        <div
          className="developer__card"
        >
          <img
            src="https://media-exp1.licdn.com/dms/image/C4D03AQGMlE1gdp-gmA/profile-displayphoto-shrink_400_400/0?e=1602115200&v=beta&t=LsP0qnf8ibaqqOxp2gcazvEmxCtwF0XsCqnUkF8N_JY"
          />
          <div>
            <h4>
              Anna Codina Aparicio
            </h4>
            <p>
              FullStack Web Developer
            </p>
          </div>
        </div>
        <div
          className="developer__card"
        >
          <img
            src="https://media-exp1.licdn.com/dms/image/C5603AQEZPiAr7_RCxA/profile-displayphoto-shrink_400_400/0?e=1602115200&v=beta&t=z8RyBtkp3SCyRZEnFBW1KU286hrCLlbxDYxbfSLyen8"
          />
          <div>
            <h4>
              Mart√≠ Amat Vila
            </h4>
            <p>
              FullStack Web Developer
            </p>
          </div>
        </div>
        <div
          className="developer__card"
        >
          <img
            src="https://media-exp1.licdn.com/dms/image/C4E03AQH5l5tJ8mWCrg/profile-displayphoto-shrink_400_400/0?e=1602115200&v=beta&t=LfwCAI3RsPfwpLb5mzuFLYhsyqGg9iJQBNUc18bVgSs"
          />
          <div>
            <h4>
              Santiago Robinet Garritano
            </h4>
            <p>
              FullStack Web Developer
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="info__link__container"
    >
      <h3
        className="info__title"
      >
        Info
      </h3>
      <a
        className="api__link"
        href="https://developer.edamam.com/es/api-recetas-edamam-documentacion"
      >
        Edamam API
      </a>
    </div>
  </footer>,
]
`;
